### ISSUE TITLE
NVDA causes duplicate letters in words typed with onscreen keyboard

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2998]
**Reported by hand2mouth on 2013-02-16 20:05**
I type using an onscreen keyboard with word prediction and also require auditory feedback. My keyboard is the freeware Virtual Keyboard 3.2 from the University of Lleida running on Windows 7 Pro 64 bit. Its word prediction automatically inserts a space after the selected word. When I choose a word prediction after typing a few letters, NVDA causes the word to appear with duplicate letters and backspaces the last letter, like this: I tytyp. I typed this message using Microsoft's onscreen keyboard, which works with NVDA but is excruciatingly slow and doesn't meet my needs. I can't afford the commercial onscreen keyboard programs. I realize that onscreen keyboard users probably aren't a big user group, but if there were a way to make NVDA compatible with other onscreen keyboards, that would be wonderful.

**Comment by leonardder** [comment]
@JosephSL: Do you have experience with on screen keyboards?

**Comment by Adriani90** [comment]
@josephsl, your ghoughts on this issue are very appreciated. Thanks.

**Comment by Adriani90** [comment]
@hand2mouth are you still having this problem with NVDA 2019.1.1?

**Comment by Adriani90** [comment]
@josephsl is this something which you could investigate in your win10 apps saddon?

**Comment by josephsl** [comment]
Hi,

The only experience I have with on-screen PC keyboards is touch keyboard in Windows 8 and later. If this can be addressed with an add-on, the correct add-on would be Enhanced Touch Gestures.

On Windows 8 and later, provided that one is using touch keyboard, one way to prevent (or minimize) this issue is turning off touch typing (available from touch interaction category provided that the computer is equipped with a touchscreen). This means users should perform one finger double tap to enter keys using the touch keyboard.

I think it would be better if the original poster can respond to this, or a Windows 7 user using the mentioned keyboard software can respond to this issue. If not, I think it might be time to put this issue to rest unless someone says no, as we haven't heard from the original poster for seven years.

Thanks.

**Comment by Adriani90** [comment]
We didn't get any updates from the initial author since many years. Meanwhile the situation might have significantly changed. I am closing this one as abandoned. Feel free to comment if you still have this issue and we can reopen.
